content/topics/java-specific/oop-basics/_index.md

```diff
@@ -1,15 +1,179 @@
 ---
 title: Java OOP basics resources and readings
+ready: true
 ---
 
-TODO:
+## OOP  in Java
+The main aim of OOP is to bind together the data and the functions that operate on them.This is done so that no other part of the code can access this data except that function.
 
-- class
-- calling functions
-- instantiation of classes
+There are four OOP priciples in java:
 
-- encapsulation
-- inheritence
-- etc...
+### Abstraction
+In Object-oriented programming, abstraction is a process of hiding the implementation details from the user, only the functionality will be provided to the user. In other words, the user will have the information on what the object does instead of how it does it.
```

Comment:
An every day example of abstraction is driving a car. When you turn on the ignition you just turn a key, the car does a whole lot of things under the hood. The starter motor and carberator is abstracted. You don't need to know how that stuff works in order to work a car.

Reply:
Added this example